Frameworks/Foundation/NSOperationQueue.mm

```diff
@@ -196,6 +196,12 @@ - (NSOperationQueue*)_doMainWork {
         didWork = FALSE;
 
         for (int i = 0; i < NSOperationQueuePriority_Count; i++) {
+            [priv->suspendedCondition lock];
+            while (priv->isSuspended) {
+                [priv->suspendedCondition wait];
+            }
```

Comment:
[http://web.stanford.edu/class/cs140/cgi-bin/lecture.php?topic=locks](http://web.stanford.edu/class/cs140/cgi-bin/lecture.php?topic=locks) may be useful reading. A while on a condition variable is a pretty typical pattern to wait until the desired condition is met. Remember that it will broadcast on any set (not just a set to false from true).

---
In reply to: [69805759](https://github.com/Microsoft/WinObjC/pull/613#discussion_r69805759) [](ancestors = 69805759)